docs/en/query_language/functions/array_functions.md

```diff
@@ -458,84 +458,143 @@ SELECT arraySlice([1, 2, NULL, 4, 5], 2, 3) AS res
 
 Array elements set to `NULL` are handled as normal values.
 
-## arraySort(arr)
+## arraySort([func,] arr, ...)
 
-Returns an array as result of sorting the elements of `arr` in ascending order.  
+Sorts the elements of the `arr` array in ascending order.  
 
-The [Schwartzian transform](https://en.wikipedia.org/wiki/Schwartzian_transform) is used to improve sorting efficiency.
-
-Example 1
+Example of integer values sorting:
 
 ``` sql
-SELECT arraySort([1, 3, 3, 0])
+SELECT arraySort([1, 3, 3, 0]);
 ```
 ```
 ┌─arraySort([1, 3, 3, 0])─┐
 │ [0,1,3,3]               │
 └─────────────────────────┘
 ```
 
-Example 2
+Example of string values sorting:
 
 ``` sql
-SELECT arraySort(['hello', 'world', '!'])
+SELECT arraySort(['hello', 'world', '!']);
 ```
 ```
 ┌─arraySort(['hello', 'world', '!'])─┐
 │ ['!','hello','world']              │
 └────────────────────────────────────┘
 ```
 
-Note that NULLs and NaNs go last (NaNs go before NULLs). For example:
- 
+Consider the following sorting order for the `NULL`, `NaN` and `Inf` values:
+
+``` sql
+SELECT arraySort([1, nan, 2, NULL, 3, nan, -4, NULL, inf, -inf]);
+```
+```
+┌─arraySort([1, nan, 2, NULL, 3, nan, -4, NULL, inf, -inf])─┐
+│ [-inf,-4,1,2,3,inf,nan,nan,NULL,NULL]                     │
+└───────────────────────────────────────────────────────────┘
+```
+
+- `-Inf` values are first in the array.
+- `NULL` values are last in the array.
+- `NaN` values are right before `NULL`.
+- `Inf` values are right before `NaN`.
+
+Note that `arraySort` is a [high-order function](higher_order_functions.md). You can pass a lambda function to it as the first argument. In this case, sorting order is determined by the result of the lambda function applied to the elements of the array. Example is shown below.
+
 ``` sql
-SELECT arraySort([1, nan, 2, NULL, 3, nan, 4, NULL])
+SELECT arraySort(lambda(tuple(x, y), y), ['hello', 'world'], [2, 1]) as res;
```

Comment:
this notation of lambda is hard to read, let's only use the short one like below